app/commands/solution/publish_iteration.rb

```diff
@@ -0,0 +1,23 @@
+class Solution
+  class PublishIteration
+    include Mandate
+
+    initialize_with :solution, :iteration_idx
+
+    def call
+      solution.update!(published_iteration: iteration, num_loc: num_loc)
+    end
+
+    private
+    memoize
+    def iteration
+      solution.iterations.find_by(idx: iteration_idx)
+    end
+
+    def num_loc
+      return solution.iterations.last.num_loc if iteration_idx.nil?
+
+      iteration.num_loc
```

Comment:
The idx might be present but invalid. And the `iterations.last` might be deleted so we need to use `latest_iteration` (which might also be nil, thus the `&`). Could you add tests accordingly pls?

```suggestion
      iteration ? iteration.num_loc : solution.latest_iteration&.num_loc
```

Reply:
Done. I also changed the logic in the `GenerateIterationSnippetJob` to use the latest iteration instead of the last: https://github.com/exercism/website/pull/2084/files#diff-578db803a141a22b493d6bc4dec4f050741e90e57b9923dc8b959f17bf8a28e7R40. Is that correct? If so, I'll also update the snippet generator job, from which I copied the existing code.

Reply:
Yes, that's correct. Good catch :)

Reply:
Updated